File: 3732-CalculateScoreAfterPerformingInstructions/3732-CalculateScoreAfterPerformingInstructions.py
# Last updated: 04/04/2026, 13:10:40
class Solution:
    def calculateScore(self, instructions: List[str], values: List[int]) -> int:
        n = len(values)

        vs = [1 for i in range(n)]
        
        i = 0
        ans = 0
        while (i >=0 and i <n ) and (vs[i] == 1):
            vs[i] =  0
            if ( instructions[i] == "add") :
                ans+=values[i]
                
                
                i+=1
            else:
                i = i+values[i]
                
        return ans
                
        

File: 3732-CalculateScoreAfterPerformingInstructions/3732-CalculateScoreAfterPerformingInstructions.cpp
// Last updated: 04/04/2026, 13:10:36
class Solution {
public:
    long long calculateScore(vector<string>& instructions, vector<int>& values) {

    int n = values.size();

        vector<int>vs(n,1);

        int i = 0 ;
        long long ans = 0;
        while ((i>=0 && i <n) && vs[i]==1){
             vs[i] = 0;
            if ( instructions[i] =="add"){
                ans+=values[i];
               
                i++;
                continue;
            }
            i= i+values[i];
            
        }
        return ans;
    }
};